framework/src/base/ScalarCoupleable.C

```diff
@@ -84,6 +84,15 @@ ScalarCoupleable::coupledScalar(const std::string & var_name, unsigned int comp)
   return getScalarVar(var_name, comp)->number();
 }
 
+Order
+ScalarCoupleable::coupledScalarOrder(const std::string & var_name, unsigned int comp)
+{
+  if (!isCoupledScalar(var_name, comp))
+    return FIRST;
```

Comment:
On Tue, Feb 23, 2016 at 2:40 PM, Yaqi <notifications@github.com> wrote:

> In framework/src/base/ScalarCoupleable.C
> <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_idaholab_moose_pull_6377-23discussion-5Fr53854823&d=BQMCaQ&c=54IZrppPQZKX9mLzcGdPfFD1hxrcB__aEkJFOKJFd00&r=3VAyeCMeLVVC6uUgEL2ffJ0a2Q62YeuUUIYP8elb-II&m=JJwc9U8DYMU5GxHVonP1QVACE2KyH58R8-O7oobBVmU&s=vv7kXCF3dL6NGI4c06w4gUz4lyTM8Uy6yYFXmEOSQLk&e=>
> :
>
> > @@ -84,6 +84,15 @@ ScalarCoupleable::coupledScalar(const std::string & var_name, unsigned int comp)
> >    return getScalarVar(var_name, comp)->number();
> >  }
> >
> > +Order
> > +ScalarCoupleable::coupledScalarOrder(const std::string & var_name, unsigned int comp)
> > +{
> > +  if (!isCoupledScalar(var_name, comp))
> > +    return FIRST;
>
> That is what test\src\auxkernels\CoupledScalarAux.C is doing. After
> looking into ScalarCoupleable::getDefaultValue(), the order here should
> be FEProblem::getMaxScalarOrder().
>

​CoupledScalarAux​ _is_ using the default value version. I am objecting the
case when users do _NOT_ use the default value. In that case this method
will return an incorrect value.



> —
> Reply to this email directly or view it on GitHub
> <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_idaholab_moose_pull_6377_files-23r53854823&d=BQMCaQ&c=54IZrppPQZKX9mLzcGdPfFD1hxrcB__aEkJFOKJFd00&r=3VAyeCMeLVVC6uUgEL2ffJ0a2Q62YeuUUIYP8elb-II&m=JJwc9U8DYMU5GxHVonP1QVACE2KyH58R8-O7oobBVmU&s=CF_WLSNJYPFTv22G9PWP0Qa1lg_wayHXp2X2uO7ItWI&e=>
> .
>